neo/Network/P2P/RemoteNode.cs

```diff
@@ -120,30 +120,31 @@ protected override void OnData(ByteString data)
         protected override void OnReceive(object message)
         {
             base.OnReceive(message);
-            switch (message)
-            {
-                case Message msg:
-                    EnqueueMessage(msg);
-                    break;
-                case IInventory inventory:
-                    OnSend(inventory);
-                    break;
-                case Relay relay:
-                    OnRelay(relay.Inventory);
-                    break;
-                case VersionPayload payload:
-                    OnVersionPayload(payload);
-                    break;
-                case MessageCommand.Verack:
-                    OnVerack();
-                    break;
-                case ProtocolHandler.SetFilter setFilter:
-                    OnSetFilter(setFilter.Filter);
-                    break;
-                case PingPayload payload:
-                    OnPingPayload(payload);
-                    break;
-            }
+            if(!base.messageProcessed)
```

Comment:
I think the performance of this `if` is the same as the `switch`. There is almost no difference.

Reply:
Igor told me the same....ehauwhahau

Reply:
@igormcoelho, please help with the final decision.

Reply:
@erikzhang, in terms of threads and assync acess, is there any issue?

Reply:
There is no assync access in actors.

Reply:
I'll quote Donald Knuth:

“The real problem is that programmers have spent far too much time worrying about efficiency in the wrong places and at the wrong times; premature optimization is the root of all evil (or at least most of it) in programming.”

Leave the switch, its much more readable and pehaps much better than ifs.